### ISSUE TITLE
Data race in Template.Parse

### ISSUE THREAD

**Issue by bep** [nicksnyder/go-i18n#181]
I get a data race reported in some of Hugo's tests:

https://travis-ci.org/gohugoio/hugo/jobs/540393974

https://github.com/nicksnyder/go-i18n/blob/master/internal/template.go#L16

**Comment by bep** [comment]
I tried to have a go at fixing this, and my first instinct would be to:

```go
t.parseInit.Do(func() {
		if strings.Contains(t.Src, leftDelim) {
			gt, err := gotemplate.New("").Funcs(funcs).Delims(leftDelim, rightDelim).Parse(t.Src)
			t.Template = gt
			t.ParseErr = err
		}
	})
```

But that doesn't work since you take `leftDelim` etc. as argument to `Parse` and hence needs to be reparsed. Locking with a mutex would work, but sounds unneededly ineffecient; why not parse the template once when you create the `Template`? Do you support multiple delimiters? Possibly related to #182

**Comment by nicksnyder** [comment]
Thanks for reporting, I will take a look!

> why not parse the template once when you create the Template?

This might be the right path forward.

**Comment by nicksnyder** [comment]
> Do you support multiple delimiters?

Technically, each translation could define its own set of delimiters, but that actually isn't the problem here I think since those delimiters would be stored in the translation file and I could lift those parameter out of the parse function.

> why not parse the template once when you create the Template?

The limiting factor here is actually `Funcs` from the `LocalizeConfig`. Since we don't know this when the templates are loaded it needs to be lazily parsed. Might need to lock; still thinking about it.

**Comment by nicksnyder** [comment]
This should be fixed in v2.0.1
